cli/crates/server/src/servers.rs

```diff
@@ -372,10 +373,10 @@ async fn validate_dependencies() -> Result<(), ServerError> {
 // to avoid issues when starting multiple CLIs simultainiously,
 // we segment the ephemeral port range into 100 segments and select a segment based on the last two digits of the process ID.
 // this allows for simultainious start of up to 100 CLIs
-fn get_bridge_port() -> Result<u16, ServerError> {
+fn get_bridge_port(http_port: u16) -> Result<u16, ServerError> {
     // must be 0-99, will fit in u16
     #[allow(clippy::cast_possible_truncation)]
-    let segment = (process::id() % 100) as u16;
+    let segment = http_port % 100;
```

Comment:
This works for serial ports like we use in tests, but it wouldn't work if e.g. you use `5000`, `6000`

Reply:
Wow, I was convinced that PR was merged...
In which case would you use 5000/6000 ports? I had regular issues with that on my Linux to be honest, tests complaining about port being already used. Otherwise maybe `http_port + process::id()`?

Reply:
I mean if a user were manually to use 5000 and 6000

Reply:
It can also happen that the two process ids have the same modulo. `http_port` is at least deterministic IMHO. In practice for most users, I would expect the cli to search for an available http_port and do 5000, 5001 etc. for the user. Now if you have a strong opinion against it, I'll change it back.

Reply:
The idea here is that processes that are launched one after another will have consecutive IDs, so the modulo colliding happens only after 100 processes

Reply:
Well, I'm not sure what you can assume from process ids, but what I'm sure is that running `cargo nextest run` I always/very often ended up with errors like: `error binding to 127.0.0.1:51763: error creating server listener: Address already in use (os error 98)`. But I didn't anymore with this change. `cargo test` seems ok though.

Reply:
That's because we use serial port IDs, but we can try this out and see if anyone runs into issues

Reply:
What do you mean by "serial port IDs"? searching sequentially for a port? If so I don't see how we could do differently when searching for a port, we just need a good offset